Fixed verify in unit tests

edwin-pixel-1 opened the pull request:
- Added verify method to view mock object
- Removed unused constant in constants class

edwin-pixel-1 committed:
Fixed verify in unit tests

- Added verify method to view mock object
- Removed unused constant in constants class

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -105,7 +105,7 @@ public void getUserInfoFromInteractorAndLoadIntoView() {
 
         OnFinishedListenerCallBack.getValue().onFinishedUser(user);
```
Use camelCase to rename `OnFinishedListenerCallBack`.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -105,7 +105,7 @@ public void getUserInfoFromInteractorAndLoadIntoView() {
         OnFinishedListenerCallBack.getValue().onFinishedUser(user);
 
-        view.showSubscriberDetails(user);
+        verify(view).showSubscriberDetails(user);
```
You should verify that the progress indicator is shown.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -127,7 +127,7 @@ public void getNullUserInfoRequestFromInteractor() {
         OnFinishedListenerCallBack.getValue().onFailureUser(errorMessage, typeError);
 
         int messageId = ErrorMessagesHelper.getMessage(typeError);
-        view.showUserError(messageId);
```
You should verify that the progress indicator is shown.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -149,7 +149,7 @@ public void getNotConnectionUserInfoRequestFromInteractor() {
         OnFinishedListenerCallBack.getValue().onFailureUser(errorMessage, typeError);
 
         int messageId = ErrorMessagesHelper.getMessage(typeError);
-        view.showUserError(messageId);
```
You should verify that the progress indicator is shown.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -173,7 +173,7 @@ public void getRepositoriesListFromInteractorAndLoadIntoView() {
 
         OnFinishedListenerCallBack.getValue().onFinishedRepository(repoList);
 
-        view.showSubscriberUserRepositories(repoList);
```
You should verify that the progress indicator is shown.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -196,7 +196,7 @@ public void getEmptyRepositoriesListFromInteractor() {
         OnFinishedListenerCallBack.getValue().onFailureRepository(errorMessage, typeError);
 
         int messageId = ErrorMessagesHelper.getMessage(typeError);
-        view.showUserError(messageId);
```
You should verify that the progress indicator is shown.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -218,7 +218,7 @@ public void getCancelledRepositoriesListRequestFromInteractor() {
         OnFinishedListenerCallBack.getValue().onFailureRepository(errorMessage, typeError);
 
         int messageId = ErrorMessagesHelper.getMessage(typeError);
-        view.showUserError(messageId);
```
You should verify that the progress indicator is shown.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java:
```diff
@@ -89,7 +89,7 @@ public void getSubcriberListFromInteractorAndLoadIntoView() {
 
         OnFinishedListenerCallBack.getValue().onResponse(subscriberList);
```
Remane it using camelCase.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java:
```diff
@@ -89,7 +89,7 @@ public void getSubcriberListFromInteractorAndLoadIntoView() {
 
         OnFinishedListenerCallBack.getValue().onResponse(subscriberList);
 
-        view.showSubscribersList(subscriberList);
```
You should verify that the progress indicator is shown.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java:
```diff
@@ -111,7 +111,7 @@ public void getNotConnectionSubscribersListRequestFromInteractor() {
         OnFinishedListenerCallBack.getValue().onFailure(errorMessage, typeError);
 
         int messageId = ErrorMessagesHelper.getMessage(typeError);
-        view.showSubscribersError(messageId);
```
You should verify that the progress indicator is shown.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java:
```diff
@@ -128,13 +128,12 @@ public void getEmptySubscribersListRequestFromInteractor() {
         OnFinishedListenerCallBack.getValue().onFailure(errorMessage, typeError);
 
         int messageId = ErrorMessagesHelper.getMessage(typeError);
-        view.showSubscribersError(messageId);
```
You should verify that the progress indicator is shown.

edwin-pixel-1 committed:
Added Progress indicator to Subscriber Detail

- Added progress indicator to subscriber detail
- Added veriry check to view mock object in Unit tests
- Added support to show swipe indicator progress at the beginning of
  Subscribers list fragment

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/presenter/SubscriberDetailPresenterImpl.java:
```diff
@@ -51,14 +51,18 @@ public SubscriberDetailPresenterImpl(SubscriberDetailView view, User userData, L
-        view.showSubscriberDetails(userData);
+        if (view != null) {
+            view.showSubscriberDetails(userData);
+            view.toggleProgressIndicator(false);
```
The natural order of the experience is: hide the loading then show the data, so, put `view.toggleProgressIndicator(false);` first then show the data.

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/presenter/SubscriberDetailPresenterImpl.java:
```diff
@@ -51,14 +51,18 @@ public SubscriberDetailPresenterImpl(SubscriberDetailView view, User userData, L
         int messageId = ErrorMessagesHelper.getMessage(type);
         if (messageId > 0 && view != null) {
             view.showUserError(messageId);
+            view.toggleProgressIndicator(false);
```
Same here.

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/presenter/SubscriberDetailPresenterImpl.java:
```diff
@@ -68,6 +72,7 @@ public void onFailureRepository(String errorMessage, ErrorMessagesHelper.TypeErr
         int messageId = ErrorMessagesHelper.getMessage(type);
         if (messageId > 0 && view != null) {
             view.showUserError(messageId);
+            view.toggleProgressIndicator(false);
```
Same.

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/presenter/SubscriberDetailPresenterImpl.java:
```diff
@@ -76,7 +81,10 @@ public void onFailureRepository(String errorMessage, ErrorMessagesHelper.TypeErr
-        view.showSubscriberUserRepositories(repositoryListData);
+        if (view != null) {
+            view.showSubscriberUserRepositories(repositoryListData);
+            view.toggleProgressIndicator(false);
```
Same.

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/SubscriberDetailFragment.java:
```diff
@@ -108,11 +110,20 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
     }
 
+    private void loadProgressIndicator(View view) {
+        progressBar = new ProgressDialog(view.getContext());
```
Why not declaring the Progress indicator in XML?

juliocatano replied:
+1

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/presenter/SubscribersListPresenterImpl.java:
```diff
@@ -71,6 +73,7 @@ public void onFailure(String errorMessage, ErrorMessagesHelper.TypeError type) {
         int messageId = ErrorMessagesHelper.getMessage(type);
         if (messageId > 0 && view != null) {
             view.showSubscribersError(messageId);
+            view.toggleProgress(false);
```
Hide the progress firts then show the error message.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -122,12 +124,14 @@ public void getNullUserInfoRequestFromInteractor() {
+        verify(view).toggleProgressIndicator(true);
 
-        OnFinishedListenerCallBack.getValue().onFailureUser(errorMessage, typeError);
+        OnFinishedListenerCallback.getValue().onFailureUser(errorMessage, typeError);
```
Remane the variable using camelCase to `onFinishedListenerCallback`.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -144,12 +148,14 @@ public void getNotConnectionUserInfoRequestFromInteractor() {
+        verify(view).toggleProgressIndicator(true);
 
-        OnFinishedListenerCallBack.getValue().onFailureUser(errorMessage, typeError);
+        OnFinishedListenerCallback.getValue().onFailureUser(errorMessage, typeError);
```
Remane the variable using camelCase to `onFinishedListenerCallback`.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -169,11 +175,13 @@ public void getRepositoriesListFromInteractorAndLoadIntoView() {
+        verify(view).toggleProgressIndicator(true);
 
-        OnFinishedListenerCallBack.getValue().onFinishedRepository(repoList);
+        OnFinishedListenerCallback.getValue().onFinishedRepository(repoList);
```
Remane the variable using camelCase to `onFinishedListenerCallback`.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -191,12 +199,14 @@ public void getEmptyRepositoriesListFromInteractor() {
+        verify(view).toggleProgressIndicator(true);
 
-        OnFinishedListenerCallBack.getValue().onFailureRepository(errorMessage, typeError);
+        OnFinishedListenerCallback.getValue().onFailureRepository(errorMessage, typeError);
```
Remane the variable using camelCase to `onFinishedListenerCallback`.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -213,12 +223,14 @@ public void getCancelledRepositoriesListRequestFromInteractor() {
+        verify(view).toggleProgressIndicator(true);
 
-        OnFinishedListenerCallBack.getValue().onFailureRepository(errorMessage, typeError);
+        OnFinishedListenerCallback.getValue().onFailureRepository(errorMessage, typeError);
```
Remane the variable using camelCase to `onFinishedListenerCallback`.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java:
```diff
@@ -85,11 +80,13 @@ public void getSubcriberListFromInteractorAndLoadIntoView() {
+        verify(view).toggleProgress(true);
 
-        OnFinishedListenerCallBack.getValue().onResponse(subscriberList);
+        OnFinishedListenerCallback.getValue().onResponse(subscriberList);
```
Remane the variable using camelCase to `onFinishedListenerCallback`.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java:
```diff
@@ -106,12 +103,14 @@ public void getNotConnectionSubscribersListRequestFromInteractor() {
+        verify(view).toggleProgress(true);
 
-        OnFinishedListenerCallBack.getValue().onFailure(errorMessage, typeError);
+        OnFinishedListenerCallback.getValue().onFailure(errorMessage, typeError);
```
Remane the variable using camelCase to `onFinishedListenerCallback`.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java:
```diff
@@ -128,13 +127,14 @@ public void getEmptySubscribersListRequestFromInteractor() {
-        verify(interactor).getSubscribersDataList(OnFinishedListenerCallBack.capture());
-
-        OnFinishedListenerCallBack.getValue().onFailure(errorMessage, typeError);
+        OnFinishedListenerCallback.getValue().onFailure(errorMessage, typeError);
```
Remane the variable using camelCase to `onFinishedListenerCallback`.

edwin-pixel-1 committed:
Fixed small suggestions

- Fixed camel case in unit tests
- Fixed the order to show the progress indicator

lucaslabs commented on app/build.gradle:
```diff
@@ -28,7 +28,8 @@ dependencies {
     compile "com.squareup.retrofit2:converter-gson:$rootProject.ext.retrofit2Version"
     compile "com.squareup.picasso:picasso:$rootProject.ext.picassoVersion"
-
+    compile "com.github.d-max:spots-dialog:$rootProject.ext.spotsDialogVersion"
```
Please keep it simple and use the progress dialog.

lucaslabs commented:
LG
